### PR TITLE
Add a new Waffle tutorial

### PR DESCRIPTION
**Pull request by duckception**
## Description
Adds a new tutorial for [Waffle](https://github.com/ethworks/waffle) about dynamic mocking and testing contract calls.

### CODE REVIEW

**Comment by samajammin** [comment]
Thanks for the submission @duckception! I've converted this PR to a "draft". Please let me know when it's complete & ready for review. Cheers!

**Comment by samajammin** [comment]
Hey @duckception - checking in on this.

**Comment by duckception** [comment]
Hey @samajammin. Sorry, it took so long. The PR is ready to be merged!

File: src/content/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+Congrats! Now you know how to use Waffle to test contract calls and mock contracts dynamically. There are far more interesting features to discover. I recommend diving into Waffle's documentation.
+
+Waffle's documentation is available [here](https://ethereum-waffle.readthedocs.io/).
+
+Source code for this tutorial can be found [here](https://github.com/EthWorks/Waffle/tree/master/examples/dynamic-mocking-and-testing-calls).
+
+Tutorials you may also be interested in:
+
+- [Testing smart contracts with Waffle](https://ethereum.org/en/developers/tutorials/testing-smart-contract-with-waffle/)
```
Comment (samajammin) [inline src/content/developers/tutorials/waffle-dynamic-mocking-and-testing-calls/index.md:294]:
```suggestion
- [Testing smart contracts with Waffle](/developers/tutorials/testing-smart-contract-with-waffle/)
```

**Review by samajammin — COMMENTED** [review]
LGTM! Made one edit.

**Review by samajammin — APPROVED** [review]

**Pull request merged**